internal/resources/ekscluster/resource_ekscluster.go

```diff
@@ -316,27 +325,211 @@ func resourceClusterInPlaceUpdate(ctx context.Context, d *schema.ResourceData, m
 		return diag.FromErr(errors.Wrapf(err, "Unable to get Tanzu Mission Control EKS cluster entry, name : %s", d.Get(NameKey)))
 	}
 
+	opsRetryTimeout := getRetryTimeout(d)
+
 	clusterSpec := constructSpec(d)
 
 	// TODO: update nodepools seperatly
+	err = handleNodepoolDiffs(config, opsRetryTimeout, getResp.EksCluster.FullName, clusterSpec.NodePools)
+	if err != nil {
```

Comment:
Since the node pool and the cluster can be updated, in the provider we should also try doing the best possible. 
In this case, we should try updating nodepool and cluster, then return error if anyone of the operation failed.

Aborting the possible cluster update below is not an ideal approach